backend/project-management-system/src/main/java/org/egov/works/util/BoundaryUtil.java

```diff
@@ -36,37 +36,48 @@ public class BoundaryUtil {
-
-        if (response.isPresent()) {
-            LinkedHashMap responseMap = (LinkedHashMap) response.get();
-            if (CollectionUtils.isEmpty(responseMap))
-                throw new CustomException("BOUNDARY ERROR", "The response from location service is empty or null");
-            String jsonString = new JSONObject(responseMap).toString();
-
-            for (String location: locations) {
-                String jsonpath = "$..boundary[?(@.code==\"{}\")]";
-                jsonpath = jsonpath.replace("{}", location);
-                DocumentContext context = JsonPath.parse(jsonString);
-                Object boundaryObject = context.read(jsonpath);
-
-                if (!(boundaryObject instanceof ArrayList) || CollectionUtils.isEmpty((ArrayList) boundaryObject)) {
-                    throw new CustomException("INVALID_BOUNDARY_DATA", "The boundary data for the code "
-                            + location + " is not available");
+    /* Takes map of locations with boundaryType as key and list of boundaries as its values, tenantId, requestInfo and hierarchyType.
+    * For each boundaryType, egov-location service is called with hierarchyType, tenantId and codes as parameters. The boundaries in request are validated against the egov-location response*/
+    public void validateBoundaryDetails(Map<String, List<String>> locations, String tenantId, RequestInfo requestInfo, String hierarchyTypeCode) {
+        for (Map.Entry<String, List<String>> entry : locations.entrySet()) {
+            String boundaryType = entry.getKey();
+            List<String> boundaries = entry.getValue();
+
+            log.info("Validating boundary for boundary type " + boundaryType + " with hierarchyType " + hierarchyTypeCode);
+            StringBuilder uri = new StringBuilder(locationHost);
+            uri.append(locationContextPath).append(locationEndpoint);
+            uri.append("?").append("tenantId=").append(tenantId);
+
+            if (hierarchyTypeCode != null)
+                uri.append("&").append("hierarchyTypeCode=").append(hierarchyTypeCode);
+
+            uri.append("&").append("boundaryType=").append(boundaryType).append("&").append("codes=")
+                    .append(StringUtils.join(boundaries, ','));
+
+            Optional<Object> response = Optional.ofNullable(serviceRequestRepository.fetchResult(uri, RequestInfoWrapper.builder().requestInfo(requestInfo).build()));
+
+            if (response.isPresent()) {
+                LinkedHashMap responseMap = (LinkedHashMap) response.get();
+                if (CollectionUtils.isEmpty(responseMap))
+                    throw new CustomException("BOUNDARY ERROR", "The response from location service is empty or null");
+                String jsonString = new JSONObject(responseMap).toString();
+
+                for (String boundary: boundaries) {
+                    String jsonpath = "$..boundary[?(@.code==\"{}\")]";
```

Comment:
This could to a constant and be part of Constants Class.
What if response is not present.